### PR TITLE
Enable Custom Pricing Provider in OpenCost for Public Cloud Kubernetes Clusters

### LINKED ISSUE #2219: Opencost auto-defines cluster as GCP and crushes due to the missing GCP key
**Describe the bug**
I use the private cloud and I have 3 instances (dev, pre-prod, prod). 
I'm trying to install opencost into all of them and I'm facing an issue with incorrect Provider set, during the installation with custom prices:
```
│ opencost 2023-10-17T09:25:18.412930604Z INF Could not find Custom Pricing file at path '/tmp/custom-config/gcp.json'                                                                                           
│ opencost 2023-10-17T09:25:18.41385985Z INF Found ProviderID starting with "gce", using GCP Provider
```
Off course, then the application crashes and being restarted via k8s. I checked my configMaps and there is an `custom-pricing-model` with `default.json` file, which is used by opencost deployment.

But I don't see any options to avoid setting "GCP" as main provider, btw, it was defined automatically.

This issue was on dev cluster, but after a few restarts this gone. On prod cluster it was installed successfully.
Clusters have no diffs in settings (just CPU and RAM, DNS names).

**To Reproduce**
Install via helm chart with custom prices.

**Expected behavior**
Installed

**Screenshots**
![image](https://github.com/opencost/opencost/assets/19505042/d6dfb9ab-1230-49db-a09f-811602c76a3f)
![image](https://github.com/opencost/opencost/assets/19505042/7a1f56ab-9995-41b1-93c0-1c6cbb55a64c)


**Which version of OpenCost are you using?**
1.106.3

**Additional context**
```
│ opencost 2023-10-17T09:18:28.351317819Z INF Could not find Custom Pricing file at path '/tmp/custom-config/gcp.json'                                                                                            │
│ opencost 2023-10-17T09:18:28.352195639Z INF Found ProviderID starting with "gce", using GCP Provider                                                                                                            │
│ opencost panic: Supply a GCP Key to start getting data                                                                                                                                                          │
│ opencost                                                                                                                                                                                                        │
│ opencost goroutine 1 [running]:                                                                                                                                                                                 │
│ opencost github.com/opencost/opencost/pkg/costmodel.Initialize({0x0, 0x0, 0xc000c5fc38?})                                                                                                                       │
│ opencost     /home/runner/work/release-scripts/release-scripts/release-scripts/workdir-prod-tag/kubecost/opencost/pkg/costmodel/router.go:1598 +0x3a2d                                                          │
│ opencost github.com/opencost/opencost/pkg/cmd/costmodel.Execute(0x1?)                                                                                                                                           │
│ opencost     /home/runner/work/release-scripts/release-scripts/release-scripts/workdir-prod-tag/kubecost/opencost/pkg/cmd/costmodel/costmodel.go:35 +0x105                                                      │
│ opencost github.com/opencost/opencost/pkg/cmd.newCostModelCommand.func1(0xc000529400?, {0x296294d?, 0x0?, 0x0?})                                                                                                │
│ opencost     /home/runner/work/release-scripts/release-scripts/release-scripts/workdir-prod-tag/kubecost/opencost/pkg/cmd/commands.go:108 +0x31                                                                 │
│ opencost github.com/spf13/cobra.(*Command).execute(0xc000529400, {0x439f1a8, 0x0, 0x0})                                                                                                                         │
│ opencost     /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:856 +0x67c                                                                                                                        │
│ opencost github.com/spf13/cobra.(*Command).ExecuteC(0xc0005ba000)                                                                                                                                               │
│ opencost     /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0x3bd                                                                                                                        │
│ opencost github.com/spf13/cobra.(*Command).Execute(...)                                                                                                                                                         │
│ opencost     /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:902                                                                                                                               │
│ opencost github.com/opencost/opencost/pkg/cmd.Execute(0x60?, {0x0, 0x0, 0x0})                                                                                                                                   │
│ opencost     /home/runner/work/release-scripts/release-scripts/release-scripts/workdir-prod-tag/kubecost/opencost/pkg/cmd/commands.go:61 +0x310                                                                 │
│ opencost main.main()                                                                                                                                                                                            │
│ opencost     /home/runner/work/release-scripts/release-scripts/release-scripts/workdir-prod-tag/kubecost/opencost/cmd/costmodel/main.go:11 +0x25  
```

### PR DESCRIPTION
**Pull request by janlauber**
## What does this PR change?
This PR introduces the ability to enforce the use of a custom pricing provider in OpenCost when deployed on GCP or other public cloud Kubernetes clusters. Previously, deploying on these platforms automatically defaulted to their respective public cloud providers, requiring API keys for pricing information. Now, users can activate a custom provider by setting `USE_CUSTOM_PROVIDER=true`, enabling custom pricing without the need for API keys.

## Does this PR relate to any other PRs?
* No

## How will this PR impact users?
* This change affects users who opt to use the custom provider feature by setting the environment variable USE_CUSTOM_PROVIDER=true in the OpenCost container.
* Allows greater flexibility and control over pricing data sources, particularly beneficial for users who prefer not to use public cloud provider API keys.

## Does this PR address any GitHub or Zendesk issues?
* Closes #2219 

## How was this PR tested?
Tested using a custom OpenCost image (`ghcr.io/janlauber/opencost:v1.109.0`) deployed in a GCP cluster, confirming the ability to utilize custom pricing with `USE_CUSTOM_PROVIDER` set to "true".

## Does this PR require changes to documentation?
* I think not, but it would be nice to set a note anywhere

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next OpenCost release? If not, why not?
*

### CODE REVIEW

**Comment by AjayTripathy** [comment]
Seems reasonable. Thanks for this PR! @thomasvn mind picking up this review when you get a moment?

**Comment by thomasvn** [comment]
Thanks @janlauber for the PR! I've added some comments to the associated Github issue.

**Comment by thomasvn** [comment]
@janlauber So the intended effect here is that if the user specifies `USE_CUSTOM_PROVIDER=true` and are not using a CSV provider, then OpenCost will default to using the pricing found in `default.json`?

**Comment by janlauber** [comment]
@thomasvn yes, this is the intended effect.

**Review by thomasvn — APPROVED** [review]
Thanks, LGTM!

**Comment by janlauber** [comment]
Any news so far?

**Review by teevans — APPROVED** [review]

File: pkg/cloud/provider/provider.go [hunk 1]
```diff
@@ -278,6 +278,16 @@ func getClusterProperties(node *v1.Node) clusterProperties {
...
 	}
+
+	// Check for custom provider settings
+	if env.IsUseCustomProvider() {
+		// Use CSV provider if set
+		if env.IsUseCSVProvider() {
+			cp.provider = kubecost.CSVProvider
+		}
+		return cp
+	}
```
Comment (michaelmdresser) [inline pkg/cloud/provider/provider.go]: @Sean-Holcomb do you see any downside to this short-circuit?

**Review by michaelmdresser — COMMENTED** [review]
This seems logical, but I'd like to hear Sean's thoughts. I also just approved this PR to run CI, so we want to make sure that passes too (I assume it will).

**Comment by janlauber** [comment]
any news here?

**Comment by mattray** [comment]
Do any @opencost/opencost-maintainers want to weigh in on this (assuming the test ges fixed)?

**Comment by janlauber** [comment]
I've fixed the code

**Comment by janlauber** [comment]
So let's get this merged?

**Review by mattray — APPROVED** [review]

**Pull request merged**